.github/workflows/check.yaml

```diff
@@ -26,18 +26,49 @@ jobs:
       working-directory: .
 
   func:
-    uses: canonical/bootstack-actions/.github/workflows/func.yaml@v2
     needs: lint-unit
+    name: functional tests
+    runs-on: ${{ matrix.runs-on }}
+    timeout-minutes: 120
     strategy:
       fail-fast: false
       matrix:
-        include:
-          - juju-channel: "3.4/stable"
-            command: "TEST_JUJU3=1 make functional"  # using TEST_JUJU3 due https://github.com/openstack-charmers/zaza/commit/af7eea953dd5d74d3d074fe67b5765dca3911ca6
-    with:
-      command: ${{ matrix.command }}
-      juju-channel: ${{ matrix.juju-channel }}
-      nested-containers: false
-      provider: "lxd"
-      python-version: "3.10"
-      timeout-minutes: 120
+        juju-channel: ["3.4/stable"]
+        runs-on:
+          - [ubuntu-latest]  # github hosted runners are amd64
+          - ['Ubuntu_ARM64_4C_16G_01']
+    steps:
+
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Setup Juju environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: "lxd"
+          juju-channel: ${{ matrix.juju-channel }}
+          charmcraft-channel: "2.x/stable"
+
+      - name: Show juju information
+        run: |
+          juju version
+          juju controllers | grep Version -A 1 | awk '{print $9}'
+
+      - name: Set zaza juju model constraints for architecture
```

Comment:
Do you still need this if you remove the machine constraint in the `base.yaml`?

Reply:
Yes unfortunately, it seems that juju defaults to amd64 even on arm64 machines

Reply:
Strange, I saw that in my another PR, the arch is default to whatever the machine support https://github.com/canonical/charm-prometheus-libvirt-exporter/actions/runs/10346737780/job/28635804387?pr=37